src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
+                api_object_synset_name = api_object_name_synset.name().split(".")[0]
+                if (
+                    search_string in api_object_synset_name
+                    or search_string_synset_name in api_object_synset_name
+                ):
+                    similar_keys.add(api_object_synset_name + "*")
+    if similar_keys:
+        for key in similar_keys:
+            _search_wildcard(key)
+    else:
+        queries = _get_close_matches_for_word_from_names(search_string, names)
+        if queries:
+            _print_search_results(queries, wildcard=True)
+
+
+def search(
+    search_string: str,
+    language: Optional[str] = "eng",
+    wildcard: Optional[bool] = False,
+    match_whole_word: Optional[bool] = False,
+    match_case: Optional[bool] = False,
+):
+    """Search for a word through the Fluent's object hierarchy.
+
+    Parameters
+    ----------
+    search_string: str
+        The word to search for. Semantic search is default.
+    language: str
+        The language for the semantic search.
+        English is default for the semantic search.
+        ISO 639-3 code of the language to be used for semantic search.
+        See `https://omwn.org/omw1.html` for the list of supported languages.
+        The default value is `eng` for English language.
+    wildcard: bool
+        Whether to use wildcard pattern. If ``True`` will match wildcard pattern based on ``fnmatch`` module and
+        will turn off semantic matching.
+    match_whole_word: bool
+        Whether to get exact match. If ``True`` will match exact string and will turn off semantic matching.
+    match_case: bool
+        Whether to match case. If ``True`` will match case-insensitive case.
+
+    Raises
+    ------
+    ValueError
+        If both ``wildcard`` and ``match_whole_word`` are ``True``.
+
+    Examples
+    --------
+    >>> import ansys.fluent.core as pyfluent
+    >>> pyfluent.search("font", match_whole_word=True)
+    >>> pyfluent.search("Font")
+    >>> pyfluent.search("iter*", wildcard=True)
+    >>> pyfluent.search("读", language="cmn")   # search 'read' in Chinese
+    The most similar API objects are:
+    <solver_session>.file.read (Command)
+    <solver_session>.file.import_.read (Command)
+    <solver_session>.mesh.surface_mesh.read (Command)
+    <solver_session>.tui.display.display_states.read (Command)
+    <meshing_session>.tui.display.display_states.read (Command)
+    """
+    if wildcard and match_whole_word:
+        raise ValueError(
+            "``wildcard`` cannot be ``True`` when ``match_whole_word`` is ``True``."
+        )
+    elif language and match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will match exact string and will turn off semantic matching.",
+            UserWarning,
+        )
+    elif match_whole_word:
+        warnings.warn(
+            "``match_whole_word=True`` will turn off wildcard matching.",
+            UserWarning,
+        )
```

Comment:
@hpohekar @seanpearsonuk Regarding the above warnings/errors, I feel it is a better design if the search function never throws error and tries to produce the best possible results when there are conflicting arguments (think of any web search engine). I think we can almost always infer the user's expectation even they've entered some incorrect combination of arguments. We can have some logic like the following:
1. If `wildcard` and `match_whole_word` are both specified to True, `match_whole_word` takes precedence as it is stricter. We explain that in a warning message.
2. If `language` is specified with `wildcard` or `match_whole_word`, we show a warning that `language` is unsupported in that case and ignored.
3. If `match_case` is specified with `wildcard`, we show a warning that `match_case` is unsupported in that case and ignored.
4. We don't need to show any warning message about turning off the semantic search when `wildcard` or `match_whole_word` are specified as that information is already part of function documenation.

Reply:
Yes, I'm also thinking the same that we should handle all precedence internally without throwing any exceptions/warnings.

Reply:
Done. Thank you.